centreon/src/Core/HostTemplate/Application/UseCase/FindHostTemplates/FindHostTemplates.php

```diff
@@ -0,0 +1,143 @@
+                && ! $this->user->hasTopologyRole(Contact::ROLE_CONFIGURATION_HOSTS_TEMPLATES_READ_WRITE)
+            ) {
+                $this->error(
+                    "User doesn't have sufficient rights to see host templates",
+                    ['user_id' => $this->user->getId()]
+                );
+                $presenter->setResponseStatus(
+                    new ForbiddenResponse(HostTemplateException::accessNotAllowed())
+                );
+
+                return;
+            }
+
+            $hostTemplates = $this->readHostTemplateRepository->findByRequestParameter($this->requestParameters);
+            $presenter->present($this->createResponse($hostTemplates));
+        } catch (\Throwable $ex) {
+            $presenter->setResponseStatus(new ErrorResponse(HostTemplateException::findHostTemplates($ex)));
+            $this->error($ex->getMessage());
+        }
+    }
+
+    /**
+     * @param HostTemplate[] $hostTemplates
+     *
+     * @return FindHostTemplatesResponse
+     */
+    private function createResponse(array $hostTemplates): FindHostTemplatesResponse
+    {
+        $response = new FindHostTemplatesResponse();
+
+        foreach ($hostTemplates as $hostTemplate) {
+            $response->hostTemplates[] = [
+                'id' => $hostTemplate->getId(),
+                'name' => $hostTemplate->getName(),
+                'alias' => $hostTemplate->getAlias(),
+                'snmpVersion' => $hostTemplate->getSnmpVersion() ? $hostTemplate->getSnmpVersion()->value : null,
+                'snmpCommunity' => $hostTemplate->getSnmpCommunity(),
+                'timezoneId' => $hostTemplate->getTimezoneId(),
+                'severityId' => $hostTemplate->getSeverityId(),
+                'checkCommandId' => $hostTemplate->getCheckCommandId(),
+                'checkCommandArgs' => $hostTemplate->getCheckCommandArgs(),
+                'checkTimeperiodId' => $hostTemplate->getCheckTimeperiodId(),
+                'maxCheckAttempts' => $hostTemplate->getMaxCheckAttempts(),
+                'normalCheckInterval' => $hostTemplate->getNormalCheckInterval(),
+                'retryCheckInterval' => $hostTemplate->getretryCheckInterval(),
+                'isActiveCheckEnabled' => $hostTemplate->isActiveCheckEnabled()->toInt(),
+                'isPassiveCheckEnabled' => $hostTemplate->isPassiveCheckEnabled()->toInt(),
+                'isNotificationEnabled' => $hostTemplate->isNotificationEnabled()->toInt(),
+                /**
+                 * TODO
+                 *  this is related to api (> 23.10) behaviour, where no options is egal to a full bitmask
+                 *  Do we keep this behaviour or do we return null ?
+                 */
+                'notificationOptions' => $hostTemplate->getNotificationOptions() !== []
+                    ? HostEvent::toBitmask($hostTemplate->getNotificationOptions())
+                    : HostEvent::getMaxBitmask(),
```

Comment:
```suggestion
                'notificationOptions' => HostEvent::toBitmask(
                    $hostTemplate->getNotificationOptions() ?: HostEvent::getMaxBitmask()
                ),
```

Reply:
This won't work, toBitmask() expect an array not an int